draft-ietf-httpbis-http2bis.xml

```diff
@@ -2934,16 +2934,29 @@ cookie: e=f
                 pseudo-header field to convey authority information, unless there is no authority
                 information to convey (in which case it MUST NOT generate :authority).
               </t>
+              <t>
+                Clients MUST NOT generate a request with a <tt>Host</tt> header field that differs
+                from the <tt>:authority</tt> pseudo-header field.  A server MAY treat a request as
```

Comment:
I'm not sure I get your reasoning for MAY here instead of something stronger. It seems that this is a security issue, so we should have stronger reasoning for optionality here. Personally I think this should be at least SHOULD, if not MUST.

Reply:
The reasoning is that old implementations won't apply this rule (so it can't be MUST) and some might choose not to (as it requires an extra comparison).  I'm OK with "SHOULD", but "MAY" seemed safer.

Reply:
IMO if there's significant security impact, it can be a MUST - especially since this is about recipient behaviour, not sender behaviour.

Reply:
I agree with @mnot. I also think that for _senders_ we can safely say that the fields MUST be byte-for-byte identical.